main/contracts/contracts.ts

```diff
@@ -14,7 +14,7 @@ export type Newable<T = any, T2 extends T = T> = new (...args: any[]) => T2;
 export type NewableConstructorInterceptor = new (...args: any[]) => IConstructionInterceptor;
 
 // More forgiving InstanceType to support instances of an Abstract Type (not Newable)
-export type InstanceOfType<T> = T extends { prototype: infer U } ? U : never;
+export type InstanceOfType<T> = T extends { prototype: infer U } ? U : T;
```

Comment:
Does this actually solve the issue? This just means you'll get `typeof MyClass` rather than `MyClass`?
I think that you need to update the tests to ensure that this is correctly typed.
Actually, with github actions it would be fairly easy to test this with multiple version of typescript. That might be useful.

Reply:
Works correctly and infering abstract or instance types.  Added some tests around it.